sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -161,8 +162,8 @@ impl CosmosClient {
         &self,
         ctx: Context,
         database_name: S,
-        options: crate::operations::create_database::Options,
-    ) -> Result<crate::operations::create_database::Response, CosmosError> {
+        options: CreateDatabaseOptions,
+    ) -> Result<CreateDatabaseResponse, CosmosError> {
```

Comment:
Why this new name? I prefer as it was before, just `Response`. 
With the module we leave choice to the SDK user: they can either write `Response` or `create_database::Response`. With `CreateDatabaseResponse` we force the verbosity on them.

Reply:
See my [comment](https://github.com/Azure/azure-sdk-for-rust/pull/279#discussion_r643558552). Namespacing doesn't really matter. If you want to make the case it's idiomatic, I defer to @JeffreyRichter but these naming conventions (namespaced or not) have been a point of contention for a while.

Reply:
Unless something truly is idiomatic, we should strive to be consistent. We are across all the SDKs by design. The idea is to make familiarity with one language SDK portable to another and open up multi-project, multi-lang solutions for all devs targeting Azure.

Reply:
Other SDKs return a Response<T> where T is the XxxResult where Xxx is the name of the operation (like CreateDatabaseResult). So, we might want to return a Response<CreateDatabaseResult>. I know this is long but it is what other languages do. 

A Response<T> has the RawResponse field in it which has the HTTP status code and the body string (as downloaded from the service) -- customers typically ignore this entirely but it is very useful for debugging scenarios. Also, the Response has the T which is for the operation's model for the deserialized result (XxxResult).

The XxxResult has fields in it for every deserialized response header and another field for the Body; the type of the Body field is the model that is specific to the operation. This contains the deserialized resource returned from the service. The body is frequently sent back to the service. For example, the customer GETs a resource, modifies the Body model and then PUTs or PATCHes the resource back up to the service. Response headers are never sent back to the service.

Reply:
Ah, ok, that weird Java feeling I had was on purpose!😄

Collapsing all the options and responses in a module (like it is now in [operations/mod.rs](https://github.com/Azure/azure-sdk-for-rust/pull/279/files#diff-3a33a87e7d35109d7ee3489b6cb76e1cd3d61c2112643d6e1728b6a9f4fe1c07)) will create a very poor autocompletion experience (you'll either import the types by hand or end up with a ton of structs, at least two for each operation). 

That said, since it has already been discussed, I'll resolve this conversation.